### PR TITLE
feat(ui): mix highlight fills and replace apply palette (YPE-5058)

### PR DESCRIPTION
**Pull request by cameronapak**
[YPE-5058](https://lifechurch.atlassian.net/browse/YPE-5058). React web only.

The grilled spec locked the six apply hexes, unmixed persist, leftover `fffe00` paint/clear, and unmixed Words of Christ. Review then changed the fill path. We no longer JS-mix against pinned `#121212` / `#232121`. Fills are CSS `color-mix` against the live `--yv-background` and `--yv-card` tokens.

## What changed
- Apply palette is `ffec5b`, `b4ffc1`, `bbf4ff`, `ffdca7`, `ffcff8`, `dfdcff`.
- Apply stays palette-only (YPE-4494). Persist the unmixed lowercase hex, no `#`.
- Reader fill is `color-mix(in srgb, #stored calc(var(--yv-highlight-mix-p) * 100%), var(--yv-background))`. Light `p = 1`. Dark `p = 0.20`. `--yv-highlight-mix-p` lives in `theme.css` and on the reader when the theme prop wins.
- Drawer dots use the same `p` against `var(--yv-card)`.
- A host override of `--yv-background` or `--yv-card` changes the fill. Default dark still looks like the old 20% mix against `#121212` (reader) / `#232121` (dots).
- Leftover `fffe00` still paints and clears. It is not an apply swatch and is not remapped.
- Words of Christ stay unmixed: `--yv-wj` is `#94000c` light / `#e4bfc2` dark.

## Out of scope
YPE-5059 (Expo), YPE-5116 (Kotlin), a black theme, remapping stored hexes, mixing WOC, persisting mixed hexes.

## Tests
1. Palette is the six. Apply rejects the old five.
2. `highlightFillColorMix` emits `color-mix` against `var(--yv-background)` or `var(--yv-card)`. Invalid hex is dropped. `--yv-highlight-mix-p` is `1` / `0.2`.
3. Verse paint and drawer dots use that helper. Theme flip updates `p`.
4. Old `fffe00` paints and clears. No apply swatch.
5. WOC text stays unmixed on a mixed fill.

## Review
Start at `packages/ui/src/lib/highlight-colors.ts` (`HIGHLIGHT_COLORS`, `highlightFillColorMix`), then `verse.tsx` and `verse-action-popover.tsx`. Mix `p` and `--yv-wj` live in `packages/core/src/styles/theme.css`.

## Visual comparison

Storybook `VerseSelection`. Before is `main`. After shots were taken on this branch against the default tokens, so dark fills still match the 20% mix on `#121212` / `#232121`. Highlighted shots apply the first swatch.

### Verse action bar, light

| Before | After |
|:------:|:-----:|
| Verse action bar in light mode before: five saturated swatches | Verse action bar in light mode after: six softer palette swatches |

### Verse action bar, dark

| Before | After |
|:------:|:-----:|
| Verse action bar in dark mode before: five swatches on the dark popover | Verse action bar in dark mode after: six swatches mixed against the card |

### Highlighted verse, light

| Before | After |
|:------:|:-----:|
| John 1:1 highlighted in light mode before with neon yellow fffe00 | John 1:1 highlighted in light mode after with palette yellow ffec5b |

### Highlighted verse, dark

| Before | After |
|:------:|:-----:|
| John 1:1 highlighted in dark mode before with translucent old yellow | John 1:1 highlighted in dark mode after with 20 percent mix on the default dark background |

### Words of Christ, John 5:8 NIV, light

Red letters stay unmixed on the fill. Before uses `--yv-red` (`#ff3d4d`). After uses `--yv-wj` (`#94000c`).

| Before | After |
|:------:|:-----:|
| John 5:8 NIV highlighted in light mode before | John 5:8 NIV highlighted in light mode after |

### Words of Christ, John 5:8 NIV, dark

Before keeps `--yv-red` on a 30% yellow overlay. After keeps `--yv-wj` (`#e4bfc2`) on the mixed fill.

| Before | After |
|:------:|:-----:|
| John 5:8 NIV highlighted in dark mode before | John 5:8 NIV highlighted in dark mode after |

<div><a href="https://cursor.com/agents/bc-04c77a96-7e41-4534-aa8f-53ad070ec61c?cursor_ref=pr_footer&cursor_cta=open_in_web"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-04c77a96-7e41-4534-aa8f-53ad070ec61c&cursor_ref=pr_footer&cursor_cta=open_in_cursor"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

[YPE-5058]: https://lifechurch.atlassian.net/browse/YPE-5058?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

<h3>Greptile Summary</h3>

The PR replaces the React web highlight palette and changes highlight rendering to token-aware CSS color mixing while preserving stored colors and legacy-highlight removal.
- Adds six apply colors and retains lowercase, unmixed persistence.
- Mixes reader fills against `--yv-background` and popover dots against `--yv-card`.
- Adds light and dark mix-strength tokens and dedicated Words of Christ colors.
- Updates tests, stories, documentation, and release notes for the new behavior.

<h3>Confidence Score: 5/5</h3>

The PR appears safe to merge.

No blocking failure remains.

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| packages/ui/src/lib/highlight-colors.ts | Defines the six-color apply palette, normalization rules, theme mix strengths, and token-aware CSS fill expressions. |
| packages/ui/src/components/verse.tsx | Replaces RGBA highlight painting with CSS color mixing and supplies the reader’s theme-specific mix-strength variable. |
| packages/ui/src/components/verse-action-popover.tsx | Updates highlight dots to mix stored colors against the live card surface token. |
| packages/core/src/styles/theme.css | Adds scoped light and dark highlight-mix and Words of Christ theme values. |
| packages/core/src/styles/bible-reader.css | Moves Words of Christ styling from the general red token to the dedicated theme-aware token. |
| .changeset/ype-5058-highlight-palette-mix.md | Records the coordinated core and UI minor release and describes the highlight presentation changes. |

<h3>Flowchart</h3>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart LR
  A[Stored lowercase highlight hex] --> B[Highlight projection]
  B --> C[Reader verse fill]
  B --> D[Popover color dot]
  P[--yv-highlight-mix-p] --> C
  P --> D
  BG[--yv-background] --> C
  CARD[--yv-card] --> D
  C --> E[CSS color-mix reader highlight]
  D --> F[CSS color-mix swatch preview]
```

<sub>Reviews (5): Last reviewed commit: ["docs: explain why highlight fills mix in..."](https://github.com/youversion/platform-sdk-react/commit/e96e10bd1cbcac983d2ba682412f9cbb0398c260) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=56683657)</sub>

<details><summary><h4>Context used (4)</h4></summary>

- Knowledge Base — [Bible HTML transformation and browser styles](https://app.greptile.com/youversion/-/custom-context/knowledge-base/youversion/platform-sdk-react/-/docs/core-html-styling.md)
- Knowledge Base — [Scripture highlighting interactions](https://app.greptile.com/youversion/-/custom-context/knowledge-base/youversion/platform-sdk-react/-/docs/ui-highlights.md)
- Knowledge Base — [Bible content widgets](https://app.greptile.com/youversion/-/custom-context/knowledge-base/youversion/platform-sdk-react/-/docs/ui-content-components.md)
- Knowledge Base — [Package delivery workflows](https://app.greptile.com/youversion/-/custom-context/knowledge-base/youversion/platform-sdk-react/-/docs/delivery-workflows.md)
</details>

### CODE REVIEW

File: packages/ui/src/components/verse.test.tsx [hunk 1]
```diff
@@ -756,6 +767,79 @@ describe('Verse.Html - Highlight fill (theme-aware, Swift parity)', () => {
   });
 });
 
+describe('Verse.Html - leftover fffe00 paint and clear', () => {
```
Comment (abharms) [inline packages/ui/src/components/verse.test.tsx]: issue: These new sibling describe blocks, plus the new case inside the existing Edge cases group in verse-action-popover.test.tsx, conflict with docs/testing.md: one optional top-level describe and no nesting for new or edited tests. Please keep the new cases flat without reshaping untouched legacy suites.

— Cursor, an AI agent sent on behalf of Austin.

File: packages/ui/src/lib/highlight-colors.ts [hunk 2]
```diff
@@ -30,13 +37,65 @@ export function normalizeHighlightHex(color: string): string | null {
...
-/** Whether `color` is one of the five SDK apply swatches (valid hex required). */
+/** Whether `color` is one of the six SDK apply swatches (valid hex required). */
 export function isPaletteHighlightColor(color: string): boolean {
   const normalized = normalizeHighlightHex(color);
   if (normalized === null) return false;
   return HIGHLIGHT_COLOR_SET.has(normalized);
 }
 
+/** `--yv-background` in light (`#ffffff`) and dark (`#121212`). Not popover `#1c1a1a`. */
+const HIGHLIGHT_SURFACE_BG = {
```
Comment (abharms) [inline packages/ui/src/lib/highlight-colors.ts]: issue (blocking): These fixed values detach highlight mixing from the semantic --yv-background / --yv-card surfaces. The README documents consumer CSS-variable overrides, so a custom dark background or card will still be mixed as if it were #121212 / #232121, producing the wrong fill. Please resolve the computed surface token, or express the mix in CSS, so the active surface remains authoritative.

— Cursor, an AI agent sent on behalf of Austin.

File: .changeset/ype-5058-highlight-palette-mix.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+'@youversion/platform-core': minor
+'@youversion/platform-react-ui': minor
+---
+
+Replace the React highlight apply palette with six colors and paint fills with CSS `color-mix` against the live `--yv-background` / `--yv-card` tokens (light identity, dark `p = 0.20`). Words of Christ stay unmixed (`#94000c` / `#e4bfc2`). Leftover `fffe00` still paints and clears.
```
Comment (cameronapak) [inline .changeset/ype-5058-highlight-palette-mix.md]: nitpick (non-blocking): This is very technical, but it's not clear about the why behind the what.

**Review by abharms — APPROVED** [review]
💯
